### PR TITLE
Adding caps to error bars

### PR DESCRIPTION
**Pull request by loichuder**
image

Part of #235 

Could have been part of #298, sorry ! :pray:

### CODE REVIEW

File: src/h5web/visualizations/line/ErrorBarCurve.tsx [hunk 1]
```diff
@@ -14,17 +14,35 @@ interface Props {
 function ErrorBarCurve(props: Props): ReactElement {
   const { abscissas, ordinates, errors, color = DEFAULT_COLOR } = props;
 
+  const capLength = 0.1 * (abscissas[1] - abscissas[0]);
```
Comment (loichuder) [inline src/h5web/visualizations/line/ErrorBarCurve.tsx:17]: Taking 10% of the spacing between two points seems to give a good aspect ratio.

**Review by axelboc — APPROVED** [review]
Great, just wondering though if we should still be working with two arrays. We could rethink the `useDataGeometry` hook so that it accepts a single array: `[x1, y1, x2, y2, ...]`. It might make things more straightforward.

- In the hook you could loop through the array with `map` and use `index % 2` to decide which scale to apply.
- In the curve components, you could `reduce` abscissas and build the array of points:

```ts
points.push(x - capLength, y - yerr, x + capLength, y - yerr); // bottom cap
points.push(x, y - yerr, x, y + yerr); // error bar
...
```

**Comment by loichuder** [comment]
> Great, just wondering though if we should still be working with two arrays. We could rethink the `useDataGeometry` hook so that it accepts a single array: `[x1, y1, x2, y2, ...]`. It might make things more straightforward.
> 
>     * In the hook you could loop through the array with `map` and use `index % 2` to decide which scale to apply.
> 
>     * In the curve components, you could `reduce` abscissas and build the array of points:
> 
> 
> ```ts
> points.push(x - capLength, y - yerr, x + capLength, y - yerr); // bottom cap
> points.push(x, y - yerr, x, y + yerr); // error bar
> ...
> ```

While handy for `ErrorBarCurve`, it will be a bit more convoluted in `DataCurve` as we would need to intertwine `abscissas` and `ordinates`.

Can we keep this for a future PR, perhaps ?

**Comment by axelboc** [comment]
Oh yeah, for sure 👍

**Pull request merged**